.github/workflows/ci.yml

```diff
@@ -0,0 +1,156 @@
+
+          files=($(git diff --name-only))
+          if [ ${#files[@]} -gt 0 ]
+          then
+            for file in $files; do
+              echo "::error file=$file::Code format check failed. To fix formatting, run 'dotnet tool restore' then 'dotnet jb cleanupcode ParquetSharp.Dataset.sln --profile=\"Built-in: Reformat Code\" --settings=ParquetSharp.Dataset.DotSettings'";
+            done
+
+            echo "Diff after formatting:"
+            git diff
+
+            exit 1
+          fi
+
+  # Build the nuget package and upload it as an artifact.
+  build-nuget:
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
+    name: Build NuGet package
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Setup .NET SDK
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: 7.0.x
+    - name: Build project
+      run: dotnet build ParquetSharp.Dataset --configuration=Release
+    - name: Build NuGet package
+      run: dotnet pack ParquetSharp.Dataset --configuration=Release --no-build --output nuget
+    - name: Upload NuGet artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: nuget-package
+        path: nuget
+
+  # Run .NET unit tests with the nuget package on all platforms and all supported .NET runtimes (thus testing the user workflow).
+  test-nuget:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        dotnet: [net6.0, net7.0]
+      fail-fast: false
+    name: Test NuGet package (.NET ${{ matrix.dotnet }} on ${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    needs: build-nuget
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Get version
+      id: get-version
+      run: echo "version=$((Select-Xml -Path ./ParquetSharp.Dataset/ParquetSharp.Dataset.csproj -XPath '/Project/PropertyGroup/Version/text()').node.Value)" >> $env:GITHUB_OUTPUT
+      shell: pwsh
+    - name: Download NuGet artifact
+      uses: actions/download-artifact@v4
+      with:
+        name: nuget-package
+        path: nuget
+    - name: Setup .NET 6 SDK
+      if: matrix.dotnet == 'net6.0'
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: 6.0.x
+    - name: Setup .NET 7 SDK
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: 7.0.x
+    - name: Add local NuGet feed
+      run: |
+        dotnet new nugetconfig
+        dotnet nuget add source -n local $PWD/nuget
+    - name: Change test project references to use local NuGet package
+      run: |
+        dotnet remove ParquetSharp.Dataset.Test reference ParquetSharp.Dataset/ParquetSharp.Dataset.csproj
+        dotnet add ParquetSharp.Dataset.Test package ParquetSharp.Dataset -v ${{ steps.get-version.outputs.version }}
```

Comment:
This is a good point. Often we will update the version in the repo and make changes before releasing that version, so there won't be a conflict, but there will also be times when the version already has a release. I thought that NuGet might have some way to prioritise sources, but from some quick research it looks like it just uses whichever one is faster, and there's no guarantee this will always be our local source.

We have the same problem with the main ParquetSharp repo and ParquetSharp.DataFrame, so should probably fix this everywhere. Maybe we need to add the commit hash to the version number when testing?

Reply:
We could use https://github.com/dotnet/Nerdbank.GitVersioning, to automate versioning. It also includes the commit hash in the non-release versions.

Reply:
This should be fixed now. I've added the commit sha as a version suffix unless a release is being created.